backend/src/Edna/Dashboard/Service.hs

```diff
@@ -98,20 +98,21 @@ newSubExperiment subExpId req = do
 -- Returns IDs of all removed measurements in the new sub-experiment and
 -- 4PL parameters.
 analyseNewSubExperiment ::
-  SubExperimentId -> NewSubExperimentReq -> Edna ([MeasurementId], Params4PL)
+  SubExperimentId -> NewSubExperimentReq -> Edna ([MeasurementId], Params4PLResp)
 analyseNewSubExperiment subExpId NewSubExperimentReq {..} = do
+  expId <- justOrThrow (DESubExperimentNotFound subExpId) =<< Q.getExperimentId subExpId
   measurements <- getMeasurements subExpId
-  liftIO $ (computeRemovedMeasurements measurements,) <$>
-    analyse4PL (computeNewPoints measurements)
+  result <- analyse4PL [(expId, computeNewPoints measurements)] >>= oneOrError "invalid analysis"
+  pure (computeRemovedMeasurements measurements, result)
   where
     computeNewPoints ::
-      [WithId 'MeasurementId MeasurementResp] -> [(Double, Double)]
+      [WithId 'MeasurementId MeasurementResp] -> [Measurement]
     computeNewPoints = mapMaybe stepActive
 
     stepActive ::
-      WithId 'MeasurementId MeasurementResp -> Maybe (Double, Double)
-    stepActive wi@WithId {..} =
-      (mrConcentration wItem, mrSignal wItem) <$ guard (not $ isRemoved wi)
+      WithId 'MeasurementId MeasurementResp -> Maybe Measurement
+    stepActive wi@WithId {wItem = MeasurementResp{..}} =
+      Measurement mrConcentration mrSignal mrIsEnabled <$ guard (not $ isRemoved wi)
```

Comment:
So you set `mIsOutlier` to `mrIsEnabled` here, it looks strange to me. This value is eventually ignored, so you can set it to arbitrary value, but overall it looks like a hack to me. If I read it correctly, you are using a type that stores more data than necessary and set unused data to arbitrary values. I think we should distinguish measurement that it just 2 values and measurement which is 2 values + flag whether it's outlier.